aptos-move/aptos-vm/src/move_vm_ext/resolver.rs

```diff
@@ -1,19 +1,68 @@
 // Copyright (c) Aptos
 // SPDX-License-Identifier: Apache-2.0
 
-use aptos_framework::natives::state_storage::StateStorageUsageResolver;
-use move_core_types::resolver::MoveResolver;
+use aptos_framework::{natives::state_storage::StateStorageUsageResolver, RuntimeModuleMetadataV1};
+use move_binary_format::errors::VMError;
+use move_core_types::{
+    account_address::AccountAddress,
+    language_storage::{ModuleId, StructTag},
+    resolver::MoveResolver,
+};
 use move_table_extension::TableResolver;
-use std::fmt::Debug;
 
 pub trait MoveResolverExt:
-    MoveResolver<Err = Self::ExtError> + TableResolver + StateStorageUsageResolver
+    MoveResolver<Err = VMError> + TableResolver + StateStorageUsageResolver
 {
-    type ExtError: Debug;
-}
+    fn get_module_metadata(&self, module_id: ModuleId) -> Option<RuntimeModuleMetadataV1>;
 
-impl<E: Debug, T: MoveResolver<Err = E> + TableResolver + StateStorageUsageResolver + ?Sized>
-    MoveResolverExt for T
-{
-    type ExtError = E;
+    fn get_resource_from_group(
+        &self,
+        address: &AccountAddress,
+        struct_tag: &StructTag,
+        resource_group: &StructTag,
+    ) -> Result<Option<Vec<u8>>, VMError>;
+
+    fn get_resource_group(&self, struct_tag: &StructTag) -> Result<Option<StructTag>, VMError> {
+        let metadata = self.get_module_metadata(struct_tag.module_id());
```

Comment:
I guess what @msmouse wanted is to always call this and decide which access path to use?